src/sql/magic.py

```diff
@@ -410,18 +418,57 @@ def interactive_execute_wrapper(**kwargs):
             else:
                 with_ = self._store.infer_dependencies(command.sql_original, args.save)
                 if with_:
-                    command.set_sql_with(with_)
-                    display.message(
-                        f"Generating CTE with stored snippets: {pretty_print(with_)}"
-                    )
+                    query_type = get_query_type(command.sql_original)
+
+                    if query_type != "SELECT":
+                        display.message_warning(
+                            f"Your query is using the following snippets: \
+{', '.join(with_)}. {IF_NOT_SELECT_MESSAGE} CTE generation is disabled"
+                        )
+                    else:
+                        command.set_sql_with(with_)
+                        display.message(
+                            f"Generating CTE with stored snippets: \
+{pretty_print(with_)}"
+                        )
                 else:
                     with_ = None
         else:
+            query_type = get_query_type(command.sql_original)
```

Comment:
you're calling get_query_type two times in two `if` branches, maybe it'd be  better to just call it at the top?

Reply:
actually, this runs when  `is_cte` is True, hence, you're guaranteed to get to this line if you're inside a CTE. do you still need to run `get_query_type`?

Reply:
We do need to check the query type even in the case where is_cte is true as the following will evaluate to a `CREATE` query type and not `SELECT`.

```
%%sql
WITH my_penguins AS (
    SELECT * FROM penguins.csv
) CREATE TABLE my_penguins_2 AS (
    SELECT * FROM my_penguins
)
```